src/main/java/tech/sourced/siva/Flag.java

```diff
@@ -1,7 +1,19 @@
 package tech.sourced.siva;
 
+/**
+ * Flags used by the index to mark a file as deleted. Other flags could be added in future versions.
```

Comment:
purpose of Flags is just to annotate the files with certain flags, not to mark as deleted, even though it's their only use right now